cmds/echo/echo.go

```diff
@@ -17,19 +17,82 @@ import (
 	"strings"
 )
 
-var nonewline = flag.Bool("n", false, "suppress newline")
+type flags struct {
+	noNewline, interpretEscapes bool
+}
+
+func escapeString(s string) (string, error) {
+	if len(s) < 1 {
+		return "", nil
+	}
+
+	s = strings.Split(s, "\\c")[0]
+	s = strings.Replace(s, "\\0", "\\", -1)
+
+	s = fmt.Sprintf("\"%s\"", s)
```

Comment:
These two lines don't make any sense to me.

what about:
s = fmt.Sprintf(s)

?

Reply:
yea that works.

Reply:
Can you add a comment in the code to explain why?